pallets/runtime/src/multisig.rs

```diff
@@ -331,6 +366,29 @@ decl_error! {
 }
 
 impl<T: Trait> Module<T> {
+    /// Private immutables
+
+    /// Add authorization for the accountKey to become a signer of multisig
+    fn unsafe_add_auth_for_signers(from: Signatory, target: Signatory, authorizer: T::AccountId) {
```

Comment:
You do not need both `from: Signatory` and `authorizer: T::AccountId` here. You can generate them from each other. looking at the rest of the functions, it will be better if you remove `from: Signatory` from this function parameter and generate it within this function.

Reply:
I am not sure about removing the `from`. Because If I removed it then in `unsafe_add_auth_for_signers()`function I need to create `from: Signatory` every time which will be a resource consumption especially for the `change_all_signers_and_sigs_required()` function which calls `unsafe_add_auth_for_signers()` in batch. 

If I remove `authorizer` then I am not able to recreate it with the `from` until unless I add some other utility function into the primitive. 

I don't find any problem with the current implementation it is clear and readable.

Reply:
You are passing by copying here so you are already spending the resources twice (First for generating `Signatory` and then copying it). Perhaps you should change the function to take reference instead of the value. That way, you won't have to copy every time the function is called.

Reply:
The signatory is not copied, Only AccountId gets copied. so for calculating the signatory, I spent resources only once and then pass the value of it. As it is present in the memory stack for that function.

Reply:
signatory is also being copied. The only difference is that signatory implements copy trait so the copy happens silently. You can pass it and AccountId as reference to avoid copying.